Fix up depthStencilResolve

sfricke-samsung opened the pull request:
closes #2311

Redone version of https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2677 (all information there)

Pending on spec changes in https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/4331

Changes from old PR

- Kept in corechecks for now for ease of feature bits
- Using new wording of spec change, can `continue` out earlier instead of ugly conditional logic used before
- `%u` to `PRIu32`

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 7038.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 5596 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 5596 passed.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 7177.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 5599 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 5599 passed.

Tony-LunarG reviewed (commented):
LGTM - Will VUID-VkRenderPassCreateInfo-fragmentDensityMapAttachment-TODO and
VUID-VkRenderPassCreateInfo2-pSubpasses-TODO be getting VUIDs soon, and should we wait for them?  Also will there be a future test for VUID-VkRenderPassCreateInfo2-pSubpasses-TODO?

Very nice cleanup!

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 9058.

sfricke-samsung commented:
@Tony-LunarG 

- rebased
- Labeled `TODO` with new VUs (they are in the 201 header, so can rebase or have this cherry picked when adding those cc @mikes-lunarg 
- Added a `VUID-VkRenderPassCreateInfo2-pSubpasses-06473` test in as well

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 5637 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 5637 passed.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 9194.

Tony-LunarG reviewed (approved):
LGTM - thanks!
